pkg/sdk/simpleschema.go

```
@@ -0,0 +1,84 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package sdk
+
+import (
+	"fmt"
+	"strings"
+
+	awssdkmodel "github.com/aws-controllers-k8s/code-generator/pkg/api"
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+	simpleschema "github.com/kro-run/kro/pkg/simpleschema"
+)
+
+// InjectSimpleSchemaShapes processes custom shapes from the top-level custom_shapes
+// section and injects them into the SDK API model.
+// Only string types are supported - all other types will cause a panic.
+func (h *Helper) InjectSimpleSchemaShapes(sdkapi *ackmodel.SDKAPI) error {
+	if len(h.cfg.CustomShapes) == 0 {
+		return nil 
+	}
+
+	for shapeName, fieldsMap := range h.cfg.CustomShapes {
+		// Convert fieldsMap to map[string]interface{} for ToOpenAPISpec
+		schemaObj := make(map[string]interface{})
+		for fieldName, fieldType := range fieldsMap {
+			schemaObj[fieldName] = fieldType
+		}
+
+		openAPISchema, err := simpleschema.ToOpenAPISpec(schemaObj)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to parse schema for shape %s: %v", shapeName, err))
```

Can we just return an error here?

---

Yes!